llama_index/retrievers/router_retriever.py

```diff
@@ -52,9 +56,21 @@ def from_defaults(
         select_multi: bool = False,
     ) -> "RouterRetriever":
         if selector is None and select_multi:
-            selector = LLMMultiSelector.from_defaults(service_context=service_context)
+            try:
+                llm = service_context.llm_predictor.llm if service_context else None
+                selector = PydanticMultiSelector.from_defaults(llm=llm)  # type: ignore
+            except ValueError:
+                selector = LLMMultiSelector.from_defaults(
+                    service_context=service_context
+                )
         elif selector is None and not select_multi:
-            selector = LLMSingleSelector.from_defaults(service_context=service_context)
+            try:
+                llm = service_context.llm_predictor.llm if service_context else None
+                selector = PydanticSingleSelector.from_defaults(llm=llm)  # type: ignore
+            except ValueError:
+                selector = LLMSingleSelector.from_defaults(
+                    service_context=service_context
+                )
```

Comment:
very nit: could we dedup this (with `router_query_engine.py`) by pulling out to a util?

Reply:
yea true, will pull this into a util